chunkify only iterates first chunk

## How to reproduce
```ts
import { assertEquals } from "https://deno.land/std@0.84.0/testing/asserts.ts"
import { chunkify } from "https://deno.land/x/iter@v3.0.0/mod.ts"

const numbers = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9]
const expected = [[0, 1], [2, 3], [4, 5], [6, 7], [8, 9]]

assertEquals([...chunkify(numbers, 2)], expected)
```

## Result
<details><summary>error message</summary>

```
Uncaught AssertionError: Values are not equal:


    [Diff] Actual / Expected


    [
      [
        0,
        1,
      ],
+     [
+       2,
+       3,
+     ],
+     [
+       4,
+       5,
+     ],
+     [
+       6,
+       7,
+     ],
+     [
+       8,
+       9,
+     ],
    ]

    at assertEquals (https://deno.land/std@0.84.0/testing/asserts.ts:214:9)
    at <anonymous>:38:1
```

</details> 

## Reason
![image](https://user-images.githubusercontent.com/54838975/236462025-dcdd22c3-fef8-4e7f-8d58-4d03b0438cf5.png)

the `return` statement is inside the for loop, only executing it once.

--- a/lib/transformers.ts
+++ b/lib/transformers.ts
@@ -260,8 +260,8 @@ export function chunkify<T>(
       if (Array.isArray(it)) {
         for (let index = 0; index < it.length; index += chunkSize) {
           yield it.slice(index, index + chunkSize);
-          return;
         }
+        return;
       }
 
       let chunk = [];

--- a/lib/transformers_test.ts
+++ b/lib/transformers_test.ts
@@ -68,6 +68,11 @@ Deno.test("chunkify", () => {
   for (const [x, y] of transformers.chunkify(stripIterable(numbers), 2)) {
     assertEquals(x + y, 2 * x + 1);
   }
+
+  const expected = [[0, 1], [2, 3], [4, 5], [6, 7], [8, 9]];
+
+  assertEquals([...transformers.chunkify(numbers, 2)], expected);
+  assertEquals([...transformers.chunkify(stripIterable(numbers), 2)], []);
 });
 
 Deno.test("rememember", () => {

